pkg/gpu/context.go

```diff
@@ -25,12 +26,16 @@ type systemContext struct {
 
 	// nvmlLib is the NVML library used to query GPU devices
 	nvmlLib nvml.Interface
+
+	// workloadmeta contains metadata about system workloads, used to get container ID
+	workloadmeta workloadmeta.Component
 }
 
-func getSystemContext(nvmlLib nvml.Interface) (*systemContext, error) {
+func getSystemContext(nvmlLib nvml.Interface, workloadmeta workloadmeta.Component) (*systemContext, error) {
```

Comment:
To discuss: is it worth it to pass here the `ProbeDependencies` struct directly, instead of each dependency by itself?